sbi/neural_nets/embedding_nets.py

```diff
@@ -31,3 +31,136 @@ def forward(self, x: Tensor) -> Tensor:
         """
         x = self.net(x)
         return x
+
+
+class CNNEmbedding(nn.Module):
+    def __init__(
+        self, input_dim: int, output_dim: int, num_fc: int = 2, num_hiddens: int = 120
+    ):
+        """multi-layer (C)NN
+            first 2 layers are convolutional, followed by fully connected layers.
+            Performing 1d convolution and max pooling with preset configs.
+        Args:
+            input_dim: Dimensionality of input.
+            num_conv: Number of con layers.
+            num_fc: Number fully connected layer, min of 2
+            num_hiddens: number of hidden dims in fc layers
+        """
+        super().__init__()
+        self.input_dim = input_dim
+        self.output_dim = output_dim
+        self.num_hiddens = num_hiddens
+
+        # configs for conv layers
```

Comment:
hm I would allow these things to vary, maybe add them as arguments to `__init__()`?